base/partr.jl

```diff
@@ -13,12 +13,17 @@ mutable struct taskheap
     taskheap() = new(SpinLock(), Vector{Task}(undef, 256), zero(Int32), typemax(UInt16))
 end
 
+mutable struct MultiQueue
+    @atomic heaps::Vector{taskheap}
+    @atomic cong_unbias::UInt32
+    const lock::SpinLock
+end
+
+MultiQueue() = MultiQueue(taskheap[], typemax(UInt32), SpinLock())
 
 # multiqueue minheap state
 const heap_d = UInt32(8)
-const heaps = [Vector{taskheap}(undef, 0), Vector{taskheap}(undef, 0)]
-const heaps_lock = [SpinLock(), SpinLock()]
-const cong_unbias = [typemax(UInt32), typemax(UInt32)]
+const multiqs = (MultiQueue(), MultiQueue())
```

Comment:
this should not be a tuple

Reply:
Do you mind explaining why? Since we are not adding thread pools, we know that there are exactly two of them ATM. This reduces the number of pointer indirections. We can of course make it extensible with yet another DCL-like pattern easily but I don't think it's a good idea to introduce complexity when not needed.